### DISCUSSION TITLE
App always in background

### DISCUSSION THREAD

**Discussion by toby63** [FossifyOrg/General-Discussion#568] [General]
I noticed that once I start the App, it stays in the background no matter what I do.

When I had a bluetooth connection with an external device the paused music suddenly started to play again, even though I thought that I had closed the App.

So here are multiple things to discuss:
1. How to really close the App? It seems that even forcing the app to close, doesn't seem to do that.
     Edit: I just tried this method once again and now it seemed to work, but still it should not be necessary to force-close the app.
3. Maybe options inside the App should exist, to either keep it open in background or close it when you close the app.

**Comment by Aga-C** [comment]
Do you mean that you see playback controls in the notification drawer even after closing the app? That's probably the system issue, because I have the same problem with numerous other apps with media playback, e.g. in NewPipe and Firefox.

**Comment by toby63** [comment]
@Aga-C Interesting, I did not notice it yet with other apps, but maybe it is also dependent on specific circumstances :thinking:.

**Comment by naveensingh** [comment]
The app isn't always running in the background, it is started by the 'system intelligence'. 

Whenever the system receives a media button event (e.g. Bluetooth) , it starts the last used media service that supports playback resumption. IIRC even when force stopped, the system will restart the app when you press play on a Bluetooth headset. It's a feature.

**Comment by toby63** [comment]
@naveensingh Thx for the info.
But I still have three open questions/issues:
1. Could this behavior be disabled (with an option for example) or is this outside of the scope of this App?
2. I still can't fully close the App in the regular way, so I would like to have that "fixed" as well.
     By that I mean, that the app should be closed and no widget (or whatever the correct term might be) should stay open, when I slide it away, just like other Apps.
2.a. My second point from above:    Maybe options inside the App should exist, to either keep it open in background or close it when you close the app.

**Reply by naveensingh** [reply]
There is a ❌ close button in the player notification, maybe you can use that?

> Could this behavior be disabled (with an option for example) or is this outside of the scope of this App?

If you meant playback resumption from the background, I don't think it can made optional. See https://github.com/androidx/media/issues/770

> I still can't fully close the App in the regular way, so I would like to have that "fixed" as well.
> By that I mean, that the app should be closed, when I slide it away, just like other Apps.
> 2.a. My second point from above: Maybe options inside the App should exist, to either keep it open in background or close it when you close the app.

If the app is playing, I would say 'the regular way' is to pause playback using the button in the notification or the app. If the app isn't playing and you still see the player notification, you can close it using the close button I mentioned above. Also, notifications like the following are created by the system, it doesn't necessarily mean the app is running in the background:

<img src="https://github.com/FossifyOrg/Music-Player/assets/36371707/8a8c527c-c289-48dd-bc28-9e93536fb9cf" width=264 /> 


The playback service is designed to be hard to kill because there were many reports of playback stopping abruptly without user action. Stopping playback when the app is cleared from the recent screen can be made optional. I'll think about adding that option.

**Note:** Frequent cold starts usually end up consuming more system resources than an app chilling in the background.

**Reply by toby63** [reply]
Thx for the info once again.
Ok I understand, it is this way by design and a problem of Android etc., then I have to use the do-not-disturb option (or whatever it is called in english) to prevent surprises.

The close symbole in the notification widget doesn't work for me (if I click it, it does nothing), but that might be a problem of my OS and Launcher.
